web/src/hashchange.js

```diff
@@ -51,6 +51,10 @@ function get_full_url(hash) {
 }
 
 function set_hash(hash) {
+    if (hash === window.location.hash) {
+        // do not add duplicate entries in browser history
```

Comment:
This should be a complete sentence with a capital letter and a period; otherwise lgtm.